### PR TITLE
Enumeration/iteration day2

### PR DESCRIPTION
**Pull request by vanzeldsilva**

### CODE REVIEW

File: iteration/exercises/spec/inject_pattern_spec.rb [hunk 1]
```diff
@@ -147,13 +149,12 @@
...
     # summing all of the averages
-
     # Your code goes here
 
     expect(final_grade).to eq(85.40)
   end
 
-  xit 'test 10' do
+  it 'test 10' do
     menu = {
```
Comment (ahmad19) [inline iteration/exercises/spec/inject_pattern_spec.rb:158]: I want you to do this on your own.

**Review by ahmad19 — CHANGES_REQUESTED** [review]
I have commented some thoughts. Take a look.

File: enumerables/exercises_1/spec/any_pattern_spec.rb [hunk 2]
```diff
@@ -39,10 +39,13 @@
...
 
-  xit 'has a multi word phrase' do
+  it 'has a multi word phrase' do
     phrases = ["Sure!", "OK.", "I have no idea.", "Really?Whatever."]
     # Your code goes here
-
+    has_multi_word_phrase = false
+    phrases.each do |phrase|
+      has_multi_word_phrase = true if 
+    end
```
[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply enumerables/exercises_1/spec/any_pattern_spec.rb:48]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/0a4a9984c33eec67cff10c92524c05f9c634b625#diff-2ed064813792b1ed30ba7da42eb94cbbf1dd97e643e6558277b87eff3c09d2ecR47

File: enumerables/exercises_1/spec/map_pattern_spec.rb [hunk 3]
```diff
@@ -65,9 +65,13 @@
...
   end
 
-  xit 'trims last letter' do
+  it 'trims last letter' do
     animals = ["dog", "cat", "mouse", "frog", "platypus"]
     # Your code goes here
+    trimmed = []
+    animals.each do |animal|
+      trimmed << animal.slice(-1)
+    end
```
[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply enumerables/exercises_1/spec/map_pattern_spec.rb:74]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/edf36be70a7b8b10fc7e516661ec7097a33f684b#diff-a3afdfc31842fd3619d5a1995f8e4fa6058d3b6beece285c1baefe828f0de323R73

File: enumerables/exercises_1/spec/reduce_pattern_spec.rb [hunk 4]
```diff
@@ -28,10 +28,11 @@
     expect(product).to eq(210)
   end
 
-  xit 'capitalizes key words in phrase' do
+  it 'capitalizes key words in phrase' do
     keywords = ["fish", "blue"]
     phrase = 'one fish two fish red fish blue fish'
     # Your code goes here
+    phrase =
```
[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply enumerables/exercises_1/spec/reduce_pattern_spec.rb]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/d967e3ce91aadc2fc8e2323bae30745afcfb5267#diff-aeae80d542d3dd35310add7aa775296c49fa79f03d7589f8770664601d2a8d72R35

File: enumerables/exercises_1/spec/reduce_pattern_spec.rb [hunk 5]
```diff
@@ -45,19 +46,18 @@
     expect(quotient).to eq(2)
   end
 
-  xit 'subtracts smallest numbers from 100' do
+  it 'subtracts smallest numbers from 100' do
```
[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply enumerables/exercises_1/spec/reduce_pattern_spec.rb:55]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/d967e3ce91aadc2fc8e2323bae30745afcfb5267#diff-aeae80d542d3dd35310add7aa775296c49fa79f03d7589f8770664601d2a8d72R59

File: enumerables/exercises_1/spec/select_pattern_spec.rb [hunk 6]
```diff
@@ -79,19 +79,23 @@
...
+      floats << number if ('%.2f' % number)
+    end
     expect(floats).to eq([1.4, 3.5, 4.9, 9.1, 8.0])
   end
 
-  xit 'arrays' do
+  it 'arrays' do
     elements = ["CAT", ["dog"], 23, [56, 3, 8], "AIMLESS", 43, "butter"]
     # Your code goes here
     expect(arrays).to eq([["dog"], [56, 3, 8]])
```
[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply enumerables/exercises_1/spec/select_pattern_spec.rb:99]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/b2950b302b2b1b67c73cfc44590616635c500786#diff-338f2ed81ee1aa4206990b80956d2e8e7dcb054e1d0279436955566c1726f308R95

File: enumerables/exercises_1/spec/basic_enums_spec.rb [hunk 7]
```diff
@@ -1,13 +1,13 @@
 RSpec.describe 'Basic Enums Test' do
   it 'solve for x' do
-    s = ''
+    s == 'x'
     5.times do
```
[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply enumerables/exercises_1/spec/basic_enums_spec.rb]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/41e4ceaf1b67c9e99daf644fa38f501b641751ce#diff-ac558eb8602b25b9a463ab290aa07b127c9e1b0f8cd71412a3c38f64b475e47eR5

File: iteration/exercises/spec/inject_pattern_spec.rb (hunk 1 above)
Comment (vanzeldsilva) [reply iteration/exercises/spec/inject_pattern_spec.rb:157]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/3abb8f7ae9f74b759b5747ae547b31dfa70e4cb0#diff-081abc5bade9588ba60ec100cac96c7796d4f852ea7eea8f957ec6a5756680caR103

File: iteration/exercises/spec/inject_pattern_spec.rb [hunk 8]
```diff
@@ -94,13 +94,15 @@
...
     elements = [["a", 1], ["b", 9], ["c", 21]]
     # Iterate over the elements array defined above
     # to find the sum of all the integers
-
     # Your Code Here
-
+    sum_of_second_values = []
+    elements.each do |element|
+      elements.flattern!
+    end
```
[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply iteration/exercises/spec/inject_pattern_spec.rb:104]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/3abb8f7ae9f74b759b5747ae547b31dfa70e4cb0#diff-081abc5bade9588ba60ec100cac96c7796d4f852ea7eea8f957ec6a5756680caR103

File: enumerables/exercises_1/spec/one_pattern_spec.rb [hunk 9]
```diff
@@ -43,9 +43,13 @@
...
   end
 
-  xit 'one double digit number' do
+  it 'one double digit number' do
     numbers = [8, 2, 10, 333, 9, 101]
     # Your code goes here
+    count = 0
+    numbers.each do |num|
+        count += 1 unless num.size == 2 
+      end
```
[the remark this answers is not in the document]
Comment (ahmad19) [reply enumerables/exercises_1/spec/one_pattern_spec.rb:52]: Here 

```
8.digits(2)
=> [0, 0, 0, 1]
```

```
333.digits(2)
=> [1, 0, 1, 1, 0, 0, 1, 0, 1]
```

`digits(2)` returns binary number which we dont need. 

What we need is the number of digits in a number. So for example: For integer 333, the number of digits are 3. Therefore:

```
numbers.each do |number|
  digits = num.digits
  count += 1 if  && digits.length > 1 && num < 3 
end
```

File: enumerables/exercises_1/spec/select_pattern_spec.rb [hunk 10]
```diff
@@ -79,19 +79,23 @@
     expect(dinosaurs).to eq(["tyrannosaurus", "achillesaurus", "qingxiusaurus"])
   end
 
-  xit 'floats' do
+  it 'floats' do
     numbers = [3, 1.4, 3.5, 2, 4.9, 9.1, 8.0]
     # Your code goes here
+    floats = []
+    numbers.each do |number|
+      floats << number if ('%.2f' % number)
```
[the remark this answers is not in the document]
Comment (ahmad19) [reply enumerables/exercises_1/spec/select_pattern_spec.rb]: I assume that you're still doing this one ?

[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply enumerables/exercises_1/spec/select_pattern_spec.rb]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/b2950b302b2b1b67c73cfc44590616635c500786#diff-338f2ed81ee1aa4206990b80956d2e8e7dcb054e1d0279436955566c1726f308R87

File: iteration/exercises/spec/inject_pattern_spec.rb (hunk 1 above)
Comment (vanzeldsilva) [reply iteration/exercises/spec/inject_pattern_spec.rb:160]: done - https://github.com/vanzeldsilva/ruby-example/pull/4/commits/6bc266d9e673e4df5a06ca874e82b78c391a6b2c#diff-081abc5bade9588ba60ec100cac96c7796d4f852ea7eea8f957ec6a5756680caR177

Comment (vanzeldsilva) [reply iteration/exercises/spec/inject_pattern_spec.rb:162]: https://github.com/vanzeldsilva/ruby-example/pull/4/commits/d4c0112eec83ef38e9376c00223fab5d1a99730b#diff-081abc5bade9588ba60ec100cac96c7796d4f852ea7eea8f957ec6a5756680caR179 - done

File: enumerables/exercises_1/spec/reject_pattern_spec.rb [hunk 11]
```diff
@@ -9,11 +9,12 @@
     expect(filtered).to eq([2, 93, 7, 1, 31, 368])
   end
 
-  xit 'removes vowels' do
+  it 'removes vowels' do
     letters = ["a", "l", "l", " ", "y", "o", "u", "r", " ", "b", "a", "s", "e", " ", "a", "r", "e", " ", "b", "e", "l", "o", "n", "g", " ", "t", "o", " ", "u", "s"]
     remaining = []
     letters.each do |letter|
       # Your code goes here
+      remaining << letter unless letter.()
```
[the remark this answers is not in the document]
Comment (vanzeldsilva) [reply enumerables/exercises_1/spec/reject_pattern_spec.rb]: https://github.com/vanzeldsilva/ruby-example/pull/4/commits/d4c0112eec83ef38e9376c00223fab5d1a99730b#diff-d22ac7bee355895bc1bddcfb07bf36ecd5f33e388fb685c270ff90b1a3565598R17 - done

**Pull request merged**